### ISSUE TITLE
Recognize custom "outfit space"-like attributes

### ISSUE THREAD

**Issue by Ferociousfeind** [endless-sky/endless-sky#6789]
### Problem Description

When attempting to outfit a highly-customized ship, one which doesn't even use outfit space, it can be a pain to keep track of the various attributes I've invented to thoroughly describe the ship's interiors. Right now, I'm using a total of seven-

- "very low diameter module" 
- "low diameter module"
- "medium diameter module"
- "high diameter module"
- "very high diameter module"
- "turret structure"
- "radiator slot"

It gets difficult to juggle these, and being able to see them just like "outfit space free: 67/192" would be extremely useful. Unfortunately, though I'd do this myself, I am very intimidated by the code, especially code that has to interact with data files directly.

### Related Issue Links

I came up with the idea myself, like a mad man at 1:30am, here: https://discord.com/channels/251118043411775489/266345072554016768/970914789603835924 (and the following short-medium conversation below)

I am not aware of anyone else attempting to do this, or asking for this. I think we as plugin creators doing this sort of nonsense have gotten along fine before.

### Desired Solution

I would like to, like ship categories or outfit categories, define any arbitrary attributes as "like outfit space", along with the unique text ("outfit space" corresponds to "outfit space free:" and "weapon capacity" corresponds to "  weapon capacity:" with the spaces to mimic indentation), and alongside the special treatment "outfit space gets".

Notice- if an outfit is under the "Ammunition" category, and has some negative attribute _**other than "outfit space"**_, a helpful hint pops ups, saying something along the line of "This is ammunition, and to install it you need to install the weapon that uses it!" even though what I am trying to install is an ammo container, and it is using one of my custom outfit-space-like attributes actually.

At the same time, I would like for any outfit-space-like attribute which is 0/0 to not be shown on the ship info display panel, I would call this aspect a "bug fix", because potentially-spoilery types of outfit space should be hidden from the player, the same way potentially-spoilery ship categories are hidden from the player (or, really, any categories not represented in that particular shipyard).

### Alternative Approaches

I cannot imagine a much better approach, but I am open to ideas... I would want any solution to be treated similarly to other categories, like the ship or outfit categories. We can change those already!

### Additional Context

![image](https://user-images.githubusercontent.com/35383563/166409216-496ac286-8027-4167-aeca-8241d5baa819.png)
![image](https://user-images.githubusercontent.com/35383563/166409248-bbc3bf20-6b35-4e90-bd65-e6dba571cf89.png)
Notice the difference in treatment of the vanilla and plugin outfits.

**Comment by Amazinite** [comment]
#4215 should help with this, but I do recognize that this is a slightly different request in that we want to see the current amount free and the capacity instead of just the current amount. Perhaps this could be a special formatting that we display, akin to the special formatting of certain attributes that I described in #6587.

**Comment by AvianGeneticist** [comment]
An option to make outfit capacities "sticky" so they're always visible would be greatly appreciated. I would like to be able to see some attributes even if they **are** full.

(I'm also making a modification to the game that would benefit from an addition like this, but I'd prefer the player be able to see "filled" attributes for the sake of consistency.)

**Comment by Ferociousfeind** [comment]
> An option to make outfit capacities "sticky" so they're always visible would be greatly appreciated. I would like to be able to see some attributes even if they **are** full.
> 
> (I'm also making a modification to the game that would benefit from an addition like this, but I'd prefer the player be able to see "filled" attributes for the sake of consistency.)

Yes, I was careful with my wording. Any "outfit space-like" attribute that every sees a positive change on a ship should appear- i.e. if one outfit adds 10 "goobeldy gock"s and another one quickly removes all 10, (and somewhere else in the files it's specified that "goobeldy gock" is an important attribute that should show up underneath "outfit space") then the outfitter aught to tell you that you've got 0/10 goobeldy gocks, yep.

If an attribute never sees a nonzero value, I would not want it to appear. If you uninstall _both_ of those two outfits, and the ship hull doesn't specify any "goobeldy gock"s, then it should vanish, because goobeldy gocks are no longer relevant to the player, they have no installed outfits that give _or_ take it.

**Comment by Ferociousfeind** [comment]
And also, yes. An advanced stats page of any sort _would_ help alleviate this problem, but what I am asking for _is_ a little different. I'm campaigning for special privileges for some attributes :P
